inspectkv/inspectkv.go

```diff
@@ -68,23 +73,80 @@ func nextIndexVals(data []interface{}) []interface{} {
 	return append(data, nil)
 }
 
-// IndexRow is the index information composed of a handle and values.
-type IndexRow struct {
+// RecordData is the record data composed of a handle and values.
+type RecordData struct {
 	Handle int64
 	Values []interface{}
 }
 
-// ScanIndexData scans the index handles and values in a limited number.
-// It returns data and the next startVals.
+// DiffRetError is an error implementation that includes a different set of records.
+type DiffRetError struct {
+	recordA *RecordData
+	recordB *RecordData
+}
+
+const resultNotExist = -1
+
+func newDiffRetError(hA, hB int64, valsA, valsB []interface{}) *DiffRetError {
+	ra := &RecordData{Handle: hA, Values: valsA}
+	rb := &RecordData{Handle: hB, Values: valsB}
+
+	return &DiffRetError{recordA: ra, recordB: rb}
+}
+
+// Error implements error Error interface.
+func (d *DiffRetError) Error() string {
+	var msgA, msgB string
+
+	if d.recordA.Handle == resultNotExist {
+		msgA = "recordA is empty"
+	} else {
+		msgA = fmt.Sprintf("recordA handle:%d, vals:%v", d.recordA.Handle, d.recordA.Values)
+	}
+	if d.recordB.Handle == resultNotExist {
+		msgB = "recordB is empty"
+	} else {
+		msgB = fmt.Sprintf("recordB handle:%d, vals:%v", d.recordB.Handle, d.recordB.Values)
+	}
+
+	return fmt.Sprintf("results are different, %s, %s", msgA, msgB)
+}
+
+// GetIndexRecordsCount returns the total number of the index records.
+func GetIndexRecordsCount(txn kv.Transaction, kvIndex kv.Index, startVals []interface{}) (int64, error) {
```

Comment:
If u use startVals as a variable, then whether it returns the total number of index records depend on what the value it is.